en/scripting/coding-setup.md

```diff
@@ -1,41 +1,39 @@
 
-前往 VS Code 的 [官方网站](https://code.visualstudio.com/)，点击首页的下载链接即可下载。
+MacOS users decompress the downloaded package and double-click **Visual Studio Code** to run it.
 
-MacOS 用户解压下载包后双击 **Visual Studio Code** 即可运行。
+Windows users should run **VSCodeUserSetup.exe** after downloading it and follow the prompts to complete the installation.
 
-Windows 用户下载后运行 **VSCodeUserSetup.exe** 按提示完成安装即可运行。
+## Smart reminder data
 
-## 智能提示数据
+When creating a project in Cocos Creator 3.x, a [tsconfig.json](tsconfig.md) file is automatically generated in the project directory, which is configured with a directory file path for code prompts. When opening the project with VS Code and writing code it will automatically prompt the Cocos Creator engine API. If the project is upgraded, the engine API will also be updated automatically.
 
-Cocos Creator 3.x 在创建项目时，项目目录下会自动生成一个 [tsconfig.json](tsconfig.md) 文件，里面配置了一个代码提示用的目录文件路径，用 VS Code 打开项目编写代码时便会自动提示 Cocos Creator 引擎 API。若项目升级，引擎 API 也会自动更新。
-
-启动 VS Code 后选择主菜单的 **File -> Open Folder...**，在弹出的对话框中选择项目根目录，也就是 `assets`、`project.json` 所在的路径。然后新建一个脚本，或者打开原有的脚本进行编辑时，就会有语法提示了。
+After starting VS Code, select **File -> Open Folder...** in the main menu, and select the project root directory in the pop-up dialog box, which is the path where `assets` and `project.json` are located. Then when creating a new script, or opening the original script for editing, there will be a syntax prompt.
 
 ![vs code](coding-setup/vscode.png)
 
-> **注意**：当使用自定义引擎，或者切换使用内置引擎/自定义引擎时，若出现 API 智能提示不更新的情况，请执行以下步骤以获得最新的代码智能提示：
+> **Note**: When using a custom engine, or switching to use the built-in engine/custom engine, if the API smart tips are not updated, please follow the steps below to get the latest code smart tips:
 >
-> 1. 删除自定义引擎中 `bin` 目录（内置引擎是 `resources\resources\3d\engine\bin` 目录）下的 `cache` 文件夹
-> 2. 在引擎目录下执行 `npm run build-declaration`
+> 1. Delete the `cache` folder under the `bin` directory in the custom engine (the built-in engine is the `resources\resources\3d\engine\bin` directory)
+> 2. Execute `npm run build-declaration` in the engine directory
 
-## 设置文件显示和搜索过滤
+## Set file display and search filter
 
-在 VS Code 的主菜单中选择 **文件（Windows）／Code（Mac）-> 首选项 -> 设置**，或者选择左下角 ![gear.png](coding-setup/gear.png) 中的 **Setting** 选项，这个操作会打开用户配置文件 **USER SETTINGS**：
+In the main menu of VS Code, select **File (Windows)/Code (Mac) -> Preferences -> Settings**, or select in the lower left corner! [gear.png](coding-setup/gear.png) **Setting** option, this operation will open the user configuration file **USER SETTINGS**:
 
 ![vs code](coding-setup/vscode-setting.png)
 
-此时在上方的搜索框中输入 **exclude** 搜索，找到 **Files: Exclude** 和 **Search: Exclude** 模块：
+Now, enter **exclude** search in the search box above and find **Files: Exclude** and **Search: Exclude** modules:
 
 ![vs code](coding-setup/vscode-exclude.png)
 
-然后点击 **添加模式** 补充以下内容中缺少的部分：
+Next, click **Add Mode** to add the missing parts in the following content:
```

Comment:
```suggestion
Next, click **Add Pattern** to add the missing parts in the following content:
```

Reply:
Please check the images when you are translating.
![image](https://user-images.githubusercontent.com/55615178/145171176-ae036e28-037c-4ee8-b73d-6a840a57b4d5.png)